### ISSUE TITLE
[kubeflow 1.8] Kubeflow 1.8 Tracking Issue

### ISSUE THREAD

**Issue by DnPlas** [kubeflow/manifests#2442]
This issue will provide high level updates of Kubeflow 1.8 release.

* [Release timeline](https://github.com/kubeflow/community/blob/master/releases/release-1.8/README.md)
* [Release team](https://github.com/kubeflow/community/blob/master/releases/release-team.md)

TODO:
- [x] Create an issue for the release timeline proposal:
[First release timeline proposal](https://github.com/kubeflow/community/pull/621)
- [x] Add a directory for 1.8 release in [releases](https://github.com/kubeflow/community/tree/master/releases)
- [x] Update the release team md file
- [x] Create a 1.8 Release Team Board in Github
- [x] Add link of the release timeline on this description

cc: @kubeflow/release-team @jbottum

**Comment by helloericsf** [comment]
After evaluating our engineering roadmap and priorities, the BentoML team has decided to pause the integration with Kubeflow Pipelines in the 1.8 release. 
We value our collaboration with Kubeflow and apologize for any disruption this causes. We believe pausing the integration temporarily is the right decision to ensure we deliver quality features. We look forward to resuming our work together in future releases.
I wanted to let you know about this decision openly and transparently. Please reach out if you have any questions or concerns. As the serving work group liaison, I do not anticipate any changes in my role or responsibilities.
cc @DnPlas

**Comment by DnPlas** [comment]
Hi community, as we approach our feature freeze (Aug 2nd) I think it is worth to ask about anything that you folks think will require more time to be completed before that date. The release team liaisons have been doing an excellent job at communicating with WG leads, but I extend the question to the rest of the community.

cc: @kubeflow/wg-automl-leads @kubeflow/wg-manifests-leads @kubeflow/wg-notebooks-leads @kubeflow/wg-pipeline-leads @jbottum

**Comment by juliusvonkohout** [comment]
@tzstoyanov will commit the istio 1.18 upgrade in https://github.com/kubeflow/manifests/pull/2455 tomorrow and i hope that the other additional rootless stuff will not be relevant for the feature freeze as written down in the PR description.

**Comment by DnPlas** [comment]
@adriangonz FYI, just so you have the dates and all information about the upcoming release.

**Comment by kimwnasptd** [comment]
Hey @DnPlas, I'd like to sincerely ask that we have a 1 week delay. The situation with Notebooks WG is the following:
1. I'd really like to include https://github.com/kubeflow/kubeflow/pull/7179, since it's the last piece to include @TobiasGoerke's effort
2. There's an issue with our CI and it's failing to build images, which we are tracking in https://github.com/kubeflow/kubeflow/issues/7226

The plan I have in mind for Notebooks WG is to evaluate as soon as possible how can we unblock the CI and do the review on the PVCViewer integration. I'm expecting this to take until this Friday, 4th August. 

Then on Monday we cut our `v1.8-branch` in the `kubeflow/kubeflow` repo and make some small PRs we need to build all the images and update the manifests. I'd expect this to take 1 day, even though the PRs are small due to the async communication.

Lastly, these are the PRs that I'd also want to finalise during the feature freeze, but am OK to not delay the release for those and cherry-pick afterwards:
* https://github.com/kubeflow/kubeflow/pull/7028
* https://github.com/kubeflow/kubeflow/pull/7220

**Comment by DnPlas** [comment]
ACK @kimwnasptd, I'll share this information with the release team.

**Comment by kimwnasptd** [comment]
@DnPlas @NohaIhab from Notebooks WG side we've:
1. Merged the PR we wanted for the Volumes UI https://github.com/kubeflow/kubeflow/pull/7179
2. Fixed the CI https://github.com/kubeflow/kubeflow/pull/7231

We'll proceed the next couple of days now with cutting the release branch and updating our manifests for the RC

**Comment by DnPlas** [comment]
@kimwnasptd thanks for the update, please keep the team posted as we are planning to finish the manifest sync next week (Wednesday).

cc: @NohaIhab

**Comment by yhwang** [comment]
@DnPlas I just created a PR to update the `Kubeflow Tekton Pipelines` manifest to 2.0.0: #2500
cc @Tomcli

**Comment by DnPlas** [comment]
Hi folks, I would like to announce that the [Kubeflow 1.8 RC.0](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.0) is out 🎉 and that we have started with manifest testing. We expect to finish this process by the end of this week (September 15th).

I'd like to encourage community members to start testing the release and provide feedback, as well as file issues if any. I would also like to remind all Distribution owners that once Manifest Testing ends we will begin Distribution Testing on September 15th (as soon as we have the results of manifest testing).  Please get your Distributions and infrastructure ready for that stage.

I also want to take the opportunity to thank all the community members who have helped with getting to this stage. Let's keep working toward a successful release!

**Comment by DavidSpek** [comment]
I'd like to mention that the issue described in https://github.com/kubeflow/pipelines/issues/8733 seems to still be broken in Pipelines 2.0.1, which is part of the Kubeflow 1.8 release. As such, a major component of Kubeflow seems to be broken in the upcoming release.

**Comment by pmuilu** [comment]
So is it so that kfp v2 will be still broken with KF 1.8? (At least #8733 is still open)

**Comment by DnPlas** [comment]
> So is it so that kfp v2 will be still broken with KF 1.8? (At least [#8733](https://github.com/kubeflow/pipelines/issues/8733) is still open)

hey @chensun @zijianjoy pinging you folks for getting more accurate information. Do you think this issue deserves more attention?

**Comment by kromanow94** [comment]
Hello, I have a few issues with 1.8 based on my tests on [Kubeflow 1.8 RC.1](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.1):

* If the Pipeline Run Steps are Pending, there is an error shown
    ```
    Cannot get MLMD objects from Metadata store.
    ```
    This is resolved after the Pod for any of the Steps is created (so if I get this right, after `metadata-writer` creates an entry in MLMD DB based on Running Pod).

![image](https://github.com/kubeflow/manifests/assets/24961770/2517a949-c4fe-484f-b4d4-aaf7bee3896c)

* The Pipeline Run Page doesn't show the `main-logs` output artifact which makes accessing logs from the Steps after the Pod was deleted not possible.
* In the Pipeline Run Page it's not possible to access the input and output artifacts due to RBAC Access Denied error. Those artifacts are visible through the `Artifacts` Page though.

![image](https://github.com/kubeflow/manifests/assets/24961770/df914c7d-5acb-4f80-a8e3-1478f3842345)

In general I feel that previous version of the KF Pipeline UI was more informative. For example, if the Pods' Step was Pending, it was possible to see the reason of Pending state in the Pipeline Run Page. 

Is there a way to enable back the `main-logs` artifact?

**Comment by Davidnet** [comment]
Tagging @Linchin for visibility, I don't know if we should create an issue in kfp, but I think kfp is working as expected ? Let me know what you think

**Comment by kromanow94** [comment]
@Davidnet from perspective of running the Steps in the Pipeline Run, I think kfp is working as expected. It's more about the dashboard, although I'm not sure if the exception `Cannot get MLMD objects from Metadata store` is expected. Is the Pipeline Run dashboard maintained by the Pipelines WG or another one?

**Comment by Linchin** [comment]
Hi @kromanow94, thank you for your feedback. I haven't completed testing rc.1 yet, but I think with my rc.0 deployment I could answer some of your questions.

> If the Pipeline Run Steps are Pending, there is an error shown `Cannot get MLMD objects from Metadata store.`

I have reproduced this error and I will investigate further into it.

> The Pipeline Run Page doesn't show the `main-logs`

This is a v1 feature that is not implemented in v2. I have created [an issue in the KFP repo](https://github.com/kubeflow/pipelines/issues/10036) about this.

> it's not possible to access the input and output artifacts due to RBAC Access Denied error.

I haven't been able to reproduce this on an rc.0 deployment, but I will double check on rc.1.

**Comment by thesuperzapper** [comment]
@kimwnasptd there are a few PRs we need to decide if we are including in 1.8:

- https://github.com/kubeflow/kubeflow/pull/7220
    - This one is VERY important, we must get it in before cutting the final 1.8!
- https://github.com/kubeflow/kubeflow/pull/6902
    - This one is small, but an important fix for some users.
- https://github.com/kubeflow/kubeflow/pull/7028 
     - This one is quite a big change, so not sure if we should introduce it now, but if it looks very ready, it might be worth it.

There are also a few others listed in https://github.com/kubeflow/kubeflow/issues/7125 that are not yet merged.

I think we should pick at least a couple of them (aside the ARM one) and try to get them in the next 1.8 RC.

**Comment by DnPlas** [comment]
> Hi @kromanow94, thank you for your feedback. I haven't completed testing rc.1 yet, but I think with my rc.0 deployment I could answer some of your questions.
> 
> > If the Pipeline Run Steps are Pending, there is an error shown `Cannot get MLMD objects from Metadata store.`
> 
> I have reproduced this error and I will investigate further into it.
> 
> > The Pipeline Run Page doesn't show the `main-logs`
> 
> This is a v1 feature that is not implemented in v2. I have created [an issue in the KFP repo](https://github.com/kubeflow/pipelines/issues/10036) about this.
> 
> > it's not possible to access the input and output artifacts due to RBAC Access Denied error.
> 
> I haven't been able to reproduce this on an rc.0 deployment, but I will double check on rc.1.

hey @Linchin , thanks for your reply. Should we consider this issue as a blocker for the release? If so, should we expect an RC2 from the pipelines WG to fix this?

**Comment by kimwnasptd** [comment]
Regarding Notebooks, we are very close to merging the following 2 and would like to ask we wait one day tops to get those in.

* https://github.com/kubeflow/kubeflow/pull/7220
* https://github.com/kubeflow/kubeflow/pull/7310

**Comment by yhwang** [comment]
@DnPlas I'd like to update kfp-tekton from 2.0.0 to 2.0.1 and here is the PR: #2545 . Thanks!

**Comment by DnPlas** [comment]
Thanks for the update @kimwnasptd, @yhwang ! I'll wait for those two to cut the next RC.

**Comment by DnPlas** [comment]
During testing, one of our users also ran into https://github.com/kubeflow/kubeflow/issues/7273. Leaving this comment for future reference as it should be fixed for the 1.8 release.
cc: @kimwnasptd @NohaIhab

**Comment by DnPlas** [comment]
Hi folks, as we are starting the bug fixing phase of the release, I'd like to make a quick update on the status.

* We currently have [RC1](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.1) and distributions have been testing it
* We are planning an RC2 that will include:
  * https://github.com/kubeflow/kubeflow/pull/7220
  * https://github.com/kubeflow/kubeflow/pull/7310
  * Fixes for https://github.com/kubeflow/kubeflow/issues/7273
  * KServe 0.11.1 (already released)
  * Potentially a fix for https://github.com/kubeflow/pipelines/issues/8733 if it is confirmed to be an issue

The date when we release RC2 depends directly on the readiness of the above.
Could we have an update for the pending PR in kubeflow/kubeflow? @kimwnasptd @thesuperzapper

**Comment by juliusvonkohout** [comment]
Maybe https://github.com/kubeflow/kubeflow/pull/7322 is interesting as well as bugfix.

**Comment by kimwnasptd** [comment]
Hey @DnPlas, the ARM PR will not be cherry-picked for this RC since the build fails on merged PRs.

For https://github.com/kubeflow/kubeflow/pull/7310 I'll try to update the PR today

**Comment by chensun** [comment]
https://github.com/kubeflow/pipelines/issues/8733 is not a block for the release. We're good from Pipelines side.

**Comment by DnPlas** [comment]
Thanks for the update @chensun !

**Comment by juliusvonkohout** [comment]
@chensun will https://github.com/kubeflow/pipelines/pull/9946 be backported for 1.8 ?
Otherwise we cannot run v2 pipelines as non-root.

**Comment by chensun** [comment]
> @chensun will [kubeflow/pipelines#9946](https://github.com/kubeflow/pipelines/pull/9946) be backported for 1.8 ? Otherwise we cannot run v2 pipelines as non-root.

I plan to cut a KFP release (2.0.2) today, and we can include that into KF 1.8

**Comment by chensun** [comment]
@DnPlas 

KFP 2.0.2 tag is out: https://github.com/kubeflow/pipelines/releases/tag/2.0.2
Can we pull its manifests into the next RC? Thanks!

**Comment by DnPlas** [comment]
Will do, thanks @chensun !

**Comment by sachdevayash1910** [comment]
> > it's not possible to access the input and output artifacts due to RBAC Access Denied error.

I have been working on setting up 1.8-rc1 as well and have observed the same things @kromanow94 mentioned above.
In addition to this, in KF <=1.7 we could also see pod ids when we clicked on a component. This was extremely helpful from a debugging and monitoring point of view. However I don't see this anymore
<img width="1111" alt="Screen Shot 2023-10-17 at 2 36 54 PM" src="https://github.com/kubeflow/manifests/assets/83288542/

If this is intentional, is there some other way of getting this info? Happy to raise an issue if needed. For reference:
<img width="1073" alt="Screen Shot 2023-10-17 at 2 38 13 PM" src="https://github.com/kubeflow/manifests/assets/83288542/0651fcb0-6ec7-4ce5-b588-b0a84700ad11">
7bdbd50f-475c-4cdc-80c2-518fc12948b7">

**Comment by juliusvonkohout** [comment]
@sachdevayash1910 you need to be on the latest 1.8 branch. There is a bug in the KFP profile controller, where they forgot to add the serviceaccount of pipelines-ui to their rolebindings. We hacked this into the normal profile controller, but it might not be in RC1.

**Comment by paravatha** [comment]
I am seeing `RBAC: access denied` even on notebooks page when I try to open a notebook instance 
_notebook/{user}/test-vscode/_

I am using https://github.com/kubeflow/manifests/tree/v1.8-branch on EKS 1.26 with oauth2-proxy

**Comment by thesuperzapper** [comment]
@paravatha which oauth2-proxy manifests are you using?

The issue will probably be related to the oauth2-proxy manifests not being updated yet to support the new security features of ensuring that only the istio gateway can talk to the notebook servers (preventing in-cluster access hacking).

See here for more info: https://github.com/kubeflow/kubeflow/pull/7310

I am not sure who is responsible for maintaining the ouath2-proxy manifests, because they are not technically officially "released" yet.

**Comment by paravatha** [comment]
@thesuperzapper 
just the alternate manifests in the same branch mentioned 
here https://github.com/kubeflow/manifests/tree/v1.8-branch#authservice
and 
here https://github.com/kubeflow/manifests/tree/v1.8-branch#dex

**Comment by DnPlas** [comment]
hey @paravatha, we identified that issue in the previous RC. Could you please try with [RC2](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.2)?

Please refer to https://github.com/kubeflow/kubeflow/pull/7310 for more information on the issue.

**Comment by DnPlas** [comment]
Hi folks,

The release team is happy to announce that we have released [Kubeflow 1.8 RC.2](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.2).

As we are approaching the release date on **October 25th**, I'd like to encourage community members to continue testing the release and provide feedback.

**Comment by paravatha** [comment]
Hi @DnPlas I tested using https://github.com/kubeflow/manifests/tree/v1.8-branch which has 1.8 rc2 changes.
It seems to be that there are 2 places `RBAC: access denied` is happening
1. on pipelines page (this may have been fixed on rc2, I have not seen it)
2. on notebooks page (I came across this on rc2, not sure of others encountered the same issue)

**Comment by chensun** [comment]
> on pipelines page (this may have been fixed in rc2, I have not seen it)

Just tested rc2, I don't see such issue.

**Comment by kimwnasptd** [comment]
Hey folks, I've seen those 2 issues and we are working on them in `1.8.0-rc.2`
1. https://github.com/kubeflow/kubeflow/issues/7373
2. https://github.com/kubeflow/kubeflow/issues/7374

**Comment by thesuperzapper** [comment]
I need a root approver for the website to approve this one, as it required updating hugo:

- https://github.com/kubeflow/website/pull/3609

**Comment by DnPlas** [comment]
Hi community,

As discussed in yesterday's community meeting, the following is expected to happen:

1. Kubeflow 1.8 GA release date Nov 1st
2. Kubeflow 1.8 RC3 and RC4 are expected to be ready today
    * [RC3](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.3) - solves all of the issues observed in Notebooks (see [this](https://github.com/kubeflow/manifests/issues/2442#issuecomment-1774694153) comment)
    * RC4 - includes Notebook servers images that support ARM architectures. See https://github.com/kubeflow/kubeflow/pull/7357 for more details.
4. Distributions and users are expected to test with RC4, but if there is any issue related to the images, we should roll back to RC3 and use that for the actual release.

As always, we encourage the community to keep testing and providing feedback. Thanks everyone for your contributions!

**Comment by thesuperzapper** [comment]
@DnPlas there was a slight CI issue with the tagging, which prevented the tags for notebook images from being created for `1.8.0-rc.4`, lets quickly resolve it by merging https://github.com/kubeflow/kubeflow/pull/7386, and then cutting `1.8.0-rc.5`.

**Comment by TristanGreathouse** [comment]
I've been experimenting with 1.8-rc1 and `kfp=2.3.0` and have run into several issues that are blockers for me and may be for others as well. A list of them is as follows:

1. Unable to use `.after()` when referencing a component within a `ParallelFor`.  [Issue](https://github.com/kubeflow/pipelines/issues/10050) with more details here.
2. Cannot assign variables from kubernetes metadata to be environment variables as we could with `kfp=1.8.21` and earlier backend versions. [Issue](https://github.com/kubeflow/pipelines/issues/10155) with more details here.
3. Cannot assign dynamic node_selector, cpu/memory requests/limits as we could with `kfp=1.8.21` and earlier backend versions. [Issue](https://github.com/kubeflow/pipelines/issues/10154) with more details here.
4. Unable to pass data artifacts from parent node outside a `ParallelFor` to a child node within a `ParallelFor`. [Issue](https://github.com/kubeflow/pipelines/issues/10149) with more details here.

Happy to provide more info if any of these are not clear!

**Comment by DnPlas** [comment]
Hi folks,

Just so you know, [RC4](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.4) is ready!

**Comment by DnPlas** [comment]
> I've been experimenting with 1.8-rc2 and `kfp=2.3.0` and have run into several issues that are blockers for me and may be for others as well. A list of them is as follows:
> 
>     1. Unable to use `.after()` when referencing a component within a `ParallelFor`.  [Issue](https://github.com/kubeflow/pipelines/issues/10050) with more details here.
> 
>     2. Cannot assign variables from kubernetes metadata to be environment variables as we could with `kfp=1.8.21` and earlier backend versions. [Issue](https://github.com/kubeflow/pipelines/issues/10155) with more details here.
> 
>     3. Cannot assign dynamic node_selector, cpu/memory requests/limits as we could with `kfp=1.8.21` and earlier backend versions. [Issue](https://github.com/kubeflow/pipelines/issues/10154) with more details here.
> 
>     4. Unable to pass data artifacts from parent node outside a `ParallelFor` to a child node within a `ParallelFor`. [Issue](https://github.com/kubeflow/pipelines/issues/10149) with more details here.
> 
> 
> Happy to provide more info if any of these are not clear!

Hi @TristanGreathouse, thanks for the feedback and for filing those issues. It sounds like a SDK issue, but I'd suggest you try RC4 to make sure you have received the latest version of pipelines (2.0.2).

Soft ping to @chensun as he is the WG lead.

**Comment by chensun** [comment]
> I've been experimenting with 1.8-rc2 and `kfp=2.3.0` and have run into several issues that are blockers for me and may be for others as well. A list of them is as follows:
> 
> 1. Unable to use `.after()` when referencing a component within a `ParallelFor`.  [Issue](https://github.com/kubeflow/pipelines/issues/10050) with more details here.
> 2. Cannot assign variables from kubernetes metadata to be environment variables as we could with `kfp=1.8.21` and earlier backend versions. [Issue](https://github.com/kubeflow/pipelines/issues/10155) with more details here.
> 3. Cannot assign dynamic node_selector, cpu/memory requests/limits as we could with `kfp=1.8.21` and earlier backend versions. [Issue](https://github.com/kubeflow/pipelines/issues/10154) with more details here.
> 4. Unable to pass data artifacts from parent node outside a `ParallelFor` to a child node within a `ParallelFor`. [Issue](https://github.com/kubeflow/pipelines/issues/10149) with more details here.
> 
> Happy to provide more info if any of these are not clear!

Thank you @TristanGreathouse for the detailed issues. 

- 1 is by design that we explicitly blocks after reference across DAGs (ParallelFor, Condition, ExitHandler are all DAGs under the hood). I do see we may be able to lift this restriction for ParallelFor, will follow up in the issue. This, if we were to change, would be SDK only change. So I think we don't need to tie it to the Kubeflow deployment release.
- 2 and 3 are both known limitations. We don't have a timeline for supporting these features at this moment, but if there's interests in contributing, we'd happy to follow up in the issues.
- 4 is a bug and I made a fix. Will do a patch release shortly. @DnPlas, I'll get back to you once this release is out. (ETA by end of this week)

**Comment by TristanGreathouse** [comment]
Thanks @DnPlas and @chensun for the responses!

- 1. Thanks, this is good to know!
- 2 & 3. Yeah I am very interested in contributing and have other members on my team that have expressed interest (CC: @sachdevayash1910). We can start looking into fixes for this, although I'm not yet super familiar with the codebase. If you already have suggestions on where to begin please let us know :)
- 4. Wow, this is great news! Thanks!

**Comment by chensun** [comment]
@DnPlas I just cut KFP 2.0.3: https://github.com/kubeflow/pipelines/releases/tag/2.0.3
Can you please update and make another RC? Thank you!

**Comment by DnPlas** [comment]
Hi everyone, [RC5](https://github.com/kubeflow/manifests/releases/tag/v1.8.0-rc.5) is here!

**Comment by yhwang** [comment]
Hi @DnPlas 
Because of KFP 2.0.3, it would be good to also update kfp-tekton to 2.0.3. I have a PR here: https://github.com/kubeflow/manifests/pull/2563

Having kfp-tekton 2.0.3 in the final RC would be great. Thanks!

**Comment by DnPlas** [comment]
@yhwang ack! I will include it in the final release.

**Comment by kromanow94** [comment]
Hi @DnPlas , great news, thanks!

Is it possible that the [[feature] display main.log as artifact for each step](https://github.com/kubeflow/pipelines/issues/10036) will be done within this release? In platforms running hundreds or thousands KF Pipelines everyday there is a high chance for a Garbage Collection on Argo WF to be configured, which means the logs are available only for as long as the Garbage Collection keeps the Pods. Also, having the possibility to see the historical execution for reference for longer than a few hours is something that's already possible in prior versions so this is a feature degradation.

**Comment by TristanGreathouse** [comment]
@DnPlas I just filed this [issue](https://github.com/kubeflow/pipelines/issues/10186) last night in which ParallelFor / Sub-DAG artifacts from component in the first DAG get overwritten by the same component (different inputs) in a duplicate sub-dag template. This basically means sub-dags cannot be repeated (albeit with different inputs/outputs) within the root dag.

**Comment by DnPlas** [comment]
> Hi @DnPlas , great news, thanks!
> 
> Is it possible that the [[feature] display main.log as artifact for each step](https://github.com/kubeflow/pipelines/issues/10036) will be done within this release? In platforms running hundreds or thousands KF Pipelines everyday there is a high chance for a Garbage Collection on Argo WF to be configured, which means the logs are available only for as long as the Garbage Collection keeps the Pods. Also, having the possibility to see the historical execution for reference for longer than a few hours is something that's already possible in prior versions so this is a feature degradation.

Hi @kromanow94 this is a feature that has its own tracking issue, but since we have already closed the call window for adding features, I'm afraid this cannot be included in this release. Pinging @Linchin and @chensun so they are aware of this for the next release cycle.

**Comment by DnPlas** [comment]
> @DnPlas I just filed this [issue](https://github.com/kubeflow/pipelines/issues/10186) last night in which ParallelFor / Sub-DAG artifacts from component in the first DAG get overwritten by the same component (different inputs) in a duplicate sub-dag template. This basically means sub-dags cannot be repeated (albeit with different inputs/outputs) within the root dag.

hey @chensun, do you mind confirming if this is a blocking issue?

**Comment by DnPlas** [comment]
Just a quick update on the RC5 testing, the release team has executed the [e2e tes](https://github.com/kubeflow/manifests/tree/master/tests/e2e) as a complementary testing and the results are positive. This plus the testing performed by distributions and users has helped to provide a better version of Kubeflow.

Here are the results of the tests:

(Tested on K8s 1.25)

```
Run ID:  5ef0267f-8b4f-45a2-8dc2-8f7ee5192ab8
Waiting for mnist-e2e experiments.kubeflow.org to get created...
mnist-e2e experiments.kubeflow.org got created.
Waiting for mnist-e2e tfjobs.kubeflow.org to get created...
mnist-e2e tfjobs.kubeflow.org got created.
Waiting for mnist-e2e tfjobs.kubeflow.org to succeed...
mnist-e2e tfjobs.kubeflow.org succeeded.
Waiting for mnist-e2e experiments.kubeflow.org to succeed...
mnist-e2e experiments.kubeflow.org succeeded.
Waiting for mnist-e2e inferenceservices.serving.kserve.io to get created...
mnist-e2e inferenceservices.serving.kserve.io got created.
Waiting for mnist-e2e inferenceservices.serving.kserve.io to succeed...
mnist-e2e inferenceservices.serving.kserve.io succeeded.
Cleaning up opened processes.
```

**Comment by DnPlas** [comment]
In preparation for the release, these are the components that must be synced into this repository:
* kubeflow/kubeflow v1.8.0 (pending GA [release](https://github.com/kubeflow/kubeflow/releases)) @kimwnasptd 
* kubeflow/pipelines v2.0.3 https://github.com/kubeflow/manifests/pull/2562
* kubeflow/katib v0.16.0 https://github.com/kubeflow/manifests/pull/2565
* kubeflow/kfp-tekton 2.0.3 https://github.com/kubeflow/manifests/pull/2563
* kubeflow/training-operator (pending GA [release](https://github.com/kubeflow/training-operator/releases)) @johnugeorge 
* kserve/kserve 0.11.1 https://github.com/kubeflow/manifests/pull/2552

**Comment by johnugeorge** [comment]
https://github.com/kubeflow/training-operator/releases/tag/v1.7.0 is created from v1.7 branch

**Comment by chensun** [comment]
> > @DnPlas I just filed this [issue](https://github.com/kubeflow/pipelines/issues/10186) last night in which ParallelFor / Sub-DAG artifacts from component in the first DAG get overwritten by the same component (different inputs) in a duplicate sub-dag template. This basically means sub-dags cannot be repeated (albeit with different inputs/outputs) within the root dag.
> 
> hey @chensun, do you mind confirming if this is a blocking issue?

Agree this is a bug, but probably not significant enough to block/delay the release. Will follow up in the issue.

**Comment by DnPlas** [comment]
Hello Kubeflow Community,

The release team is happy to announce that [Kubeflow 1.8](https://github.com/kubeflow/manifests/releases/tag/v1.8.0) is now available. The release highlights can be found in the [Kubeflow 1.8 blog post](https://blog.kubeflow.org/kubeflow-1.8-release/), and the specifics about components and their versions are in the [Kubeflow 1.8 website entry](https://www.kubeflow.org/docs/releases/kubeflow-1.8/).

We'd like to thank everyone in the community for supporting during this release cycle by planning, developing, testing, and providing feedback.It would've not been possible without the support of such great community.

If you'd like to provide feedback on the release cycle, we will have retrospective sessions in the next two Release Team Meetings at 18:00 CEST every Monday ([Community Calendar](https://www.kubeflow.org/docs/about/community/#kubeflow-community-calendars)).

Kubeflow 1.8 Release Team

**Comment by szymek116** [comment]
> @DnPlas
> 
> KFP 2.0.2 tag is out: https://github.com/kubeflow/pipelines/releases/tag/2.0.2 Can we pull its manifests into the next RC? Thanks!

Hi!
I see this was inlcided in 1.8 finally but for me still does not fix the issue, when running non rooted image I still get:

`F1112 19:54:25.274010 34 main.go:49] failed to execute component: unable to create directory "/minio/mlpipeline/v2/artifacts/hello-pipeline/605cef82-4331-4d0b-8f34-888485a3bb77/html-visualization" for output artifact "html_artifact": mkdir /minio: permission denied
`

Is anything more needed for this to be fixed ? I saw open issue on that but pretty much the same stuff that was merged at least around file permissions on that folder which I believe was root cause

https://github.com/kubeflow/pipelines/pull/6530

I see on 1.9 agenda (https://github.com/kubeflow/kubeflow/issues/6662) point try out kfp v2 rootless, I guess that's what I have done :) I can provide more details on my test, but maybe create a separate ticket for this ?

**Comment by DnPlas** [comment]
hi @szymek116, I will let @chensun confirm this information as it is very specific to pipelines. Chen, it would be great if you could also confirm if this will require a patch release from the release team.

**Comment by rvadim** [comment]
Hi! Confirm problems with permissions for `Run of [Tutorial] DSL` at a rootless env.
```
ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: '/.local'
Check the permissions.
I1115 09:26:26.725677      36 launcher_v2.go:151] publish success.
F1115 09:26:26.725728      36 main.go:49] failed to execute component: exit status 1
time="2023-11-15T09:26:27.224Z" level=info msg="sub-process exited" argo=true error="<nil>"
Error: exit status 1
time="2023-11-15T09:26:28.153Z" level=info msg="sub-process exited" argo=true error="<nil>"
Error: exit status 1
```

**Comment by chensun** [comment]
> > @DnPlas
> > KFP 2.0.2 tag is out: https://github.com/kubeflow/pipelines/releases/tag/2.0.2 Can we pull its manifests into the next RC? Thanks!
> 
> Hi! I see this was inlcided in 1.8 finally but for me still does not fix the issue, when running non rooted image I still get:
> 
> `F1112 19:54:25.274010 34 main.go:49] failed to execute component: unable to create directory "/minio/mlpipeline/v2/artifacts/hello-pipeline/605cef82-4331-4d0b-8f34-888485a3bb77/html-visualization" for output artifact "html_artifact": mkdir /minio: permission denied `
> 
> Is anything more needed for this to be fixed ? I saw open issue on that but pretty much the same stuff that was merged at least around file permissions on that folder which I believe was root cause
> 
> [kubeflow/pipelines#6530](https://github.com/kubeflow/pipelines/pull/6530)
> 
> I see on 1.9 agenda ([kubeflow/kubeflow#6662](https://github.com/kubeflow/kubeflow/issues/6662)) point try out kfp v2 rootless, I guess that's what I have done :) I can provide more details on my test, but maybe create a separate ticket for this ?

Rootless kfp is a proposal and contribution from the community that the Pipeline WG agreed and reviewed. Regrettably, it did not reach completion. While we are still open to community assistance in finishing this feature, in that case we can consider a patch release. However, in the absence of community contribution and considering our existing resource constraints and priorities, there's no timeline for shipping this feature.

**Comment by DnPlas** [comment]
@rvadim that sounds like a pipelines example specific issue, do you mind filing an issue in kubeflow/pipelines so that it gets more visibility?

cc: @chensun @Davidnet

**Comment by DnPlas** [comment]
Closing this issue as the release cycle for 1.8 has come to an end. Thanks to the community that participated by testing and providing feedback during the various phases of the release.
